src/test/java/wolox/training/controllers/UserControllerTest.java

```diff
@@ -0,0 +1,179 @@
+package wolox.training.controllers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import wolox.training.models.Book;
+import wolox.training.models.User;
+import wolox.training.repositories.BookRepository;
+import wolox.training.repositories.UserRepository;
+
+import static org.hamcrest.Matchers.hasSize;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.hamcrest.Matchers.is;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(UserController.class)
+public class UserControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private UserRepository mockedUserRepo;
+
+    @MockBean
+    private BookRepository mockedBookRepo;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    private String BASE_USERS_URL = "/api/users/";
+
+    private User javier;
+    private User marihe;
+    private User robertV;
+    Book theMist;
+
+    @BeforeEach
+    void setUp(){
+        javier = new User("JaviMo", "Javier Moreno", LocalDate.parse("1980-10-08"));
+        marihe = new User("MariHe", "Maria Haize", LocalDate.parse("1990-11-04"));
+        robertV = new User("RobertVv", "Robert Velvet", LocalDate.parse("1975-04-11"));
+        theMist = new Book("Terror", "Stephen King", "The Mist", "no value",
+            "SOME PUBLISHER", "2000", 123, "143565786", "imageOfBook");
+
+    }
+
+    List<User> allUsers = new ArrayList<User>();
+
+    @Test
+    public void givenUsers_whenGetUsers_thenReturnJsonArray() throws Exception {
+        allUsers.add(javier);
+        given(mockedUserRepo.findAll()).willReturn(allUsers);
+
+        mockMvc.perform(get(BASE_USERS_URL + "all"))
+            .andExpect(status().isOk())
+            .andExpect(content().contentType("application/json"))
```

Comment:
Maybe you should put "MediaType.APPLICATION_JSON"

Reply:
it doesn't compile directly and if i try to cast it to (ResultMatcher), it raises an execution error